### DISCUSSION TITLE
IO management with External Assets?

### DISCUSSION THREAD

**Discussion by zyd14** [dagster-io/dagster#19844] [Q&A]
It seems like External Assets are meant to replace SourceAssets, but I don't see any way to associate an external asset with an IO manager. Is there any way to have dagster-managed (meaning managed by an IOManager) loading of external assets? Or is IO left to be done within the asset definition for downstream assets? To me it seemed like the primary reason for using SourceAssets was to enable managing the loading of those assets through an IO manager, but external assets seem to not include this functionality at all.

**Comment by ion-elgreco** [comment]
That's a huge gap, and should be addressed before source assets get deprecated

**Comment by RenaudLN** [comment]
Has there be any progress on this topic?

**Reply by easontm** [reply]
It's not included in the docs, but `AssetSpec` has had `with_io_manager_key` for several months now.
